ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/msck/MsckAnalyzer.java

```diff
@@ -87,7 +87,8 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     }
 
     if (repair && AcidUtils.isTransactionalTable(table)) {
-      outputs.add(new WriteEntity(table, WriteType.DDL_EXCLUSIVE));
+      outputs.add(new WriteEntity(table, AcidUtils.isLocklessReadsEnabled(table, conf) ? 
```

Comment:
Maybe a little late, maybe it is not viable, but I feel that we spread this checks all around the code now. Would it be possible to translate the `WriteType` to the lock type in one place instead?

Maybe in a different PR later?

Reply:
probably we could try to make this translation in AcidUtils.makeLockComponents, however, there might be many hidden stones, better to try it in diff PR